core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -344,6 +348,10 @@ public boolean index(Index index) {
         } catch (OutOfMemoryError | IllegalStateException | IOException t) {
             maybeFailEngine("index", t);
             throw new IndexFailedEngineException(shardId, index.type(), index.id(), t);
+        } finally {
+            if (index.seqNo() != -1) {
```

Comment:
This is an example where I think that using the field [proposed previously](https://github.com/elastic/elasticsearch/pull/14651/files#r45225330) might clarify things in the sense that it will be easier to search for all usages of that field rather than `-1` or `-1L`.